tsconfig.json

```diff
@@ -4,9 +4,12 @@
     "baseUrl": "./src",
     "declaration": true,
     "esModuleInterop": true,
-    "lib": ["dom", "es2020"],
+    "lib": ["DOM", "ES2022"],
+    "module": "NodeNext",
```

Comment:
Does this mean it outputs ESM?

Reply:
nope, we're doing that in #427! we're still emitting CJS in this PR, but this just gives us support for the `exports` object in the `package.json` when loading in external packages. you can read more at these links:

https://www.typescriptlang.org/tsconfig/#node16nodenext-nightly-builds

https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing